components/x-interaction/src/InteractionRender.jsx

```diff
@@ -1,21 +1,13 @@
 import { h } from '@financial-times/x-engine';
-import getComponentName from './concerns/get-component-name';
 import { loading } from './concerns/symbols';
-import { addSerialisationData } from './concerns/serialiser';
-import shortId from '@quarterto/short-id';
 
 export const InteractionRender = ({
+	id,
 	actions,
 	state,
 	initialState,
+	[loading]: isLoading,
 	Component,
-	id = `${getComponentName(Component)}-${shortId()}`,
-	hydrating = false
-}) => {
-	const { [loading]: isLoading, ...props } = state || initialState;
-	addSerialisationData({ id, Component, props });
-
-	const rendered = <Component {...props} actions={actions} isLoading={isLoading} />;
-
-	return hydrating ? rendered : <div data-x-dash-id={id}>{rendered}</div>;
-};
+}) => (
+	<Component {...initialState} {...state} {...{ id, actions, isLoading }} />
```

Comment:
I think this can be simply `{...state, id, actions, isLoading }`

Reply:
`initialState` needs to be in there because it's not actually included in the `state`; doing `{...{...initialState, ...state, id, actions, isLoading}}` would mean extra object cloning (i think spread into component props doesn't clone, but spread into an object definitely does) plus it looks weird idk

Reply:
Ahh, yes, JSX spread is not the spread you were looking for. Does this look better?

```jsx
<Component {...initialState} {...state} id={id} actions={actions} isLoading={isLoading} />
```

Reply:
it used to be that, i prefer the spread version?